app/models/thredded/null_user.rb

```diff
@@ -34,6 +40,8 @@ def thredded_anonymous?
       true
     end
 
+    alias_method :anonymous?, :thredded_anonymous?
```

Comment:
ll the methods thredded ever calls on a User object should be prefixed with `thredded_` so there is no chance of conflict, as `anonymous?` may mean something else for a particular application.

> The issue with this is that the implementer does not own that NullUser class. Therefore the parent application controller might be sending methods to an object it does not control.

Thredded `NullUser` should only ever be exposed to `Thredded` controllers, not the parent app controllers. I think there is an issue in the parent application (I can have a look once you push it to github).

> Also - asking the parent class to lean heavily on a namedspaced method in a dependency feels a little weird, no?

The parent class should never need to use this method.